File: src/my_project/easy_problems/from101to150/shortest_distance_character.py
from typing import List 

class Solution:
    def shortestToChar(self, s: str, c: str) -> List[int]:
        temp = None
        solution: List[int] = list()
        index_target: List[int] = list()
        
        

        for i in range(len(s)):
            if s[i] == c:
                index_target.append(i)
             
        for i in range(len(s)):
            temp = min([abs(x - i) for x in index_target])            
            solution.append(temp)

        return solution

solution = Solution()

print(solution.shortestToChar(s = "loveleetcode", c = "e"))

File: my_project_typescript/easy_problems_typescript/from1to50/shortest_distance_character.ts
function shortestToChar(s: string, c: string): number[] {

    let indexTarget: Array<number> = [];
    let solution: Array<number> = [];
    let temp: any = null;
    let tempArray: Array<number> = [];

    function getArrayMin(array: Array<number>){
        return Math.min.apply(null, array);
     }

    for (let i = 0; i < s.length; i++){
        if (s[i] === c){
            indexTarget.push(i)
        }
    }
    
    for (let i = 0; i <  s.length; i++){
        tempArray = []
        for (let j of indexTarget) {
            tempArray.push(Math.abs(i-j))           
        }

        solution.push(getArrayMin(tempArray))
        
    }

    return solution;

};



console.log(shortestToChar( "loveleetcode", "e"))




